examples/webgl_materials_cubemap_mipmaps.html

```diff
@@ -0,0 +1,186 @@
+
+				for ( var level = 0; level <= maxLevel; ++ level ) {
+
+					var size = Math.pow( 2, maxLevel - level );
+					var imageByteLength = size * size * 4;
+					var textures = [];
+					
+					if ( byteOffset >= arrayBuffer.byteLength ) {
+
+						break;
+
+					}
+
+					for ( var face = 0; face < 6; ++ face ) {
+
+						var imageData = new Uint8Array( arrayBuffer, byteOffset, imageByteLength );
+						var texture = new THREE.DataTexture( imageData, size, size, THREE.RGBAFormat, THREE.UnsignedByteType );
+						textures.push( texture );
+						byteOffset += imageByteLength;
+					}
+
+					var levelCubeTexture = new THREE.CubeTexture( textures );
+					cubeTextures.push( levelCubeTexture );
+				}
+
+				var customizedCubeTexture = cubeTextures[0];
+				customizedCubeTexture.minFilter = THREE.LinearMipMapLinearFilter;
+				customizedCubeTexture.magFilter = THREE.LinearFilter;
+				customizedCubeTexture.format = THREE.RGBAFormat;
+				customizedCubeTexture.encoding = THREE.LogLuvEncoding;
+
+				if ( cubeTextures.length > 0 ) {
+
+					customizedCubeTexture.mipmaps = cubeTextures;
+					customizedCubeTexture.generateMipmaps = false;
+
+				}
+
+				customizedCubeTexture.needsUpdate = true;
+
+				return customizedCubeTexture;
+
+			}
+
+			function init() {
+
+				container = document.createElement( 'div' );
+				document.body.appendChild( container );
+
+				camera = new THREE.PerspectiveCamera( 50, window.innerWidth / window.innerHeight, 1, 10000 );
+				camera.position.z = 1500;
+
+				//load cubemap
+
+				scene = new THREE.Scene();
+
+				//lights
+				var ambient = new THREE.AmbientLight( 0xffffff, 1.0 );
+				scene.add( ambient );
+
+				var pendings = [];
+				loadCubeTexture( 'textures/cube/angus/specular_cubemap_256_luv.bin', 256 ).then( function ( reflectCubeTexture ) {
+
+					//rewrite mipmaps
+					var originalCubeTexture = reflectCubeTexture.clone();
+					originalCubeTexture.generateMipmaps = true;
+					originalCubeTexture.needsUpdate = true;
+
+					//models
+					var sphere = new THREE.SphereBufferGeometry( 100, 128, 128 );
+
+					var material = new THREE.MeshPhongMaterial( { color: 0xffffff, envMap: reflectCubeTexture } );
+
+					var offset = 250;
+					var originX = -1150;
+					for ( var i = 0; i < 10; ++ i ) {
+
+						//customize mipmaps sphere
+						var roughness = 0.6 * i / 10 + 0.4;
+						var cm = material.clone();
+						cm.envMap = originalCubeTexture;
+
+						var c = new THREE.Mesh( sphere, cm );
+						c.position.set( originX + i * offset, 150, 0 );
+						scene.add( c );
+
+						//webgl mipmaps sphere
+						var om = material.clone();
+
+						var o = new THREE.Mesh( sphere, om );
+						o.position.set( originX + i * offset, -150, 0 );
+						scene.add( o );
+					}
+
+				} );
+
+				//renderer
+				renderer = new THREE.WebGLRenderer( { antialias: true } );
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setClearColor( 0xf1f3f5 );
```

Comment:
When using this clear color, you have to change the color of the description text. Otherwise the text is hard to read.

BTW: The examples mostly assign a color value to `Scene.background` instead of using `WebGLRenderer.setClearColor()`.

Reply:
remove background color